server/src/lib/model/user-client.ts

```diff
@@ -262,43 +261,25 @@ const UserClient = {
     invite: string
   ): Promise<boolean> {
     if (email && challenge && team) {
-      let client_id = await Promise.all([UserClient.findClientId(email)]);
+      const client_id = (await Promise.all([
+        UserClient.findClientId(email),
+      ]))[0];
       if (client_id) {
-        // Generate a random, anonymized, and unique invite ID for enrollments.
-        // HACK(riley/can): Can we move this entirely into the DB?
-        let inviteToken;
-        while (true) {
-          inviteToken = (await hash(`${Math.random()}`)).slice(0, 12);
-          const res = await db.query(
-            `SELECT * FROM enrollments WHERE url_token=?`,
-            inviteToken
-          );
-          if (!res || !res.length) break;
-        }
-
+        // if invite is null, the invitation comes from team; OTHERWISE from user
+        // for user invitation, invite_by is the same value of url_token from another row
+        // for team invitation, invite_by is the same value of url_token of the same row
+        const [[{ team_id, challenge_id, uuid }]] = await db.query(
+          `SELECT t.id AS team_id, t.challenge_id, UUID() AS uuid FROM teams t WHERE t.url_token=?`,
+          [team]
+        );
         const res = await db.query(
           `
-          SET @challengeId := (SELECT id FROM challenges WHERE url_token=?);
-
-          INSERT IGNORE INTO enrollments (
-            id,
-            challenge_id,
-            team_id,
-            client_id,
-            invited_by,
-            url_token
-          )
-          VALUES (
-            null,
-            @challengeId,
-            (SELECT id FROM teams WHERE challenge_id=@challengeId AND url_token=?),
-            ?,
-            ?,
-            ?
-          )`,
-          [challenge, team, client_id, invite, inviteToken]
+          INSERT INTO enroll (challenge_id, team_id, client_id, invited_by, url_token)
+          VALUES (${challenge_id}, ${team_id}, '${client_id}', ?, '${uuid}')
```

Comment:
Why are some of these surrounded by `''` and some aren't?

Reply:
I put most of the values inside the sql string, only invite_by as prepared parameters.
string parameters are surrounded by ' ', numbers are not.
I see it already creates confusion. I will make it unified.

Reply:
Ah, makes sense. That's why I shouldn't do reviews on Friday evening. Thanks 😅